src/Uno.UWP/Devices/Geolocation/Geolocator.wasm.cs

```diff
@@ -0,0 +1,251 @@
+﻿#if __WASM__
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Uno;
+using Windows.Foundation;
+using static Uno.Foundation.WebAssemblyRuntime;
+
+namespace Windows.Devices.Geolocation
+{
+	public sealed partial class Geolocator
+	{		
+		private const string JsType = "Windows.Devices.Geolocation.Geolocator";
+		private const uint Wgs84SpatialReferenceId = 4326;
+
+		private static List<TaskCompletionSource<GeolocationAccessStatus>> _pendingAccessRequests = new List<TaskCompletionSource<GeolocationAccessStatus>>();
+		private static ConcurrentDictionary<string, TaskCompletionSource<Geoposition>> _pendingGeopositionRequests = new ConcurrentDictionary<string, TaskCompletionSource<Geoposition>>();
+		private static ConcurrentDictionary<string, Geolocator> _positionChangedSubscriptions = new ConcurrentDictionary<string, Geolocator>();		
+
+		private string _positionChangedRequestId;
+
+		public Geolocator()
+		{
+		}
+
+		partial void OnActualDesiredAccuracyInMetersChanged()
+		{
+			if (_positionChanged != null)
+			{
+				//reset position changed watch to apply accuracy
+				StopPositionChanged();
+				StartPositionChanged();
+			}
+		}
+
+		partial void StartPositionChanged()
+		{
+			BroadcastStatus(PositionStatus.Initializing); //GPS is initializing
+			_positionChangedRequestId = Guid.NewGuid().ToString();
+			_positionChangedSubscriptions.TryAdd(_positionChangedRequestId, this);
+			var command = $"{JsType}.startPositionWatch({ActualDesiredAccuracyInMeters},\"{_positionChangedRequestId}\")";
+			InvokeJS(command);
+		}
+
+		partial void StopPositionChanged()
+		{
+			_positionChangedSubscriptions.TryRemove(_positionChangedRequestId, out var _);
+			var command = $"{JsType}.startPositionWatch(\"{_positionChangedRequestId}\")";
+			InvokeJS(command);
+		}
+
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
+		{
+			var accessRequest = new TaskCompletionSource<GeolocationAccessStatus>();
+			lock (_pendingAccessRequests)
+			{
+				//enqueue request
+				_pendingAccessRequests.Add(accessRequest);
+			}
+			var command = $"{JsType}.requestAccess()";
```

Comment:
Sorry Jerome, I am a bit confused - what active request tracking do you mean here? They are tracked in the `_pendingAccessRequests`, or do you mean something else? My thought now is that I probably could implement it in a way to just queue waiting requests and resolve them all at once when the first one returns. The current implementation also processes the whole queue at once, but still makes multiple requests, which is probably not necessary.

Reply:
Improved, now JS request is created only when there are no pending in the queue

Reply:
By active requests, I'm talking about active access to geolocation requests. If multiple parts of the code are requesting it, may there be multiple active prompts?

Reply:
That should be covered by the code - if multiple threads trigger the geolocation access request, thanks to the lock it will be run in JS only once and all those waiting in "queue" be handled at once when the JS callback occurs, by the foreach which sets result of all pending `TaskCompletionSources`